### PR TITLE
feat: gitops actions pull requests for GitLab

### PR DESCRIPTION
**Pull request by adityathebe**
resolves: https://github.com/flanksource/mission-control/issues/791

merge after: https://github.com/flanksource/mission-control/pull/816

### CODE REVIEW

**Comment by adityathebe** [comment]
Note: go-git client doesn't support azure devops repository as of now
GH Issue: https://github.com/cerbos/cerbos/issues/1279

**Comment by moshloop** [comment]
@adityathebe - We can use go-git for AzureDevops

**Comment by adityathebe** [comment]
> @adityathebe - We can use go-git for AzureDevops

go-git cannot clone it https://github.com/go-git/go-git/issues/64

**Comment by adityathebe** [comment]
Created test PR here: https://dev.azure.com/flanksource/Demo1/_git/infrastructure/pullrequest/18

File: pkg/clients/git/connectors/connectors.go [hunk 1]
```diff
@@ -63,21 +67,45 @@ func NewConnector(gitConfig *GitopsAPISpec) (Connector, error) {
...
-			return nil, errors.Errorf("invalid repository url: %s", gitConfig.Repository)
+			return nil, errors.Errorf("invalid GitHub repository url: %s. expected: https://github.com/<owner>/<repo>", gitConfig.Repository)
 		}
 		owner := parts[0]
 		repoName := parts[1]
 		repoName = strings.TrimSuffix(repoName, ".git")
-		githubToken := gitConfig.GITHUB_TOKEN
-		return NewGithub(owner, repoName, githubToken)
+		return NewAccessTokenClient("github", owner, repoName, gitConfig.AccessToken)
+	} else if strings.HasPrefix(gitConfig.Repository, "https://gitlab.com") {
```
Comment (moshloop) [inline pkg/clients/git/connectors/connectors.go:76]: Many gitlab instances are private so we can't rely on the domain

**Pull request merged**